site/docs/v1/tech/identity-providers/twitch.adoc

```diff
@@ -0,0 +1,104 @@
+* <<Building Your Own Integration>>
+
+{empty} +
+
+Ensure you have a Twitch account, and then navigate to the Twitch Developer dashboard. Please note that you will need to have verified your account and enabled two-factor authentication.
+
+* link:https://dev.twitch.tv/[Twitch - Getting Started]
+
+== Create a Twitch Account
+
+If you do not yet have a Twitch app you will need to create an app in this step. If you already have a Twitch app that you plan to use for FusionAuth please review this section anyway to ensure your configuration is correct to support the Twitch login.
+
+* https://dev.twitch.tv/console/apps
+
+In the following screenshot you will find an example of creating an application in Twitch. Review the following sections to ensure your configuration is correct. Fill out any required fields found on this form and click the `Create` button to create a new Twitter app.
+
+
+=== Add the Callback URL
+
+In this example I have registered `\https://login.piedpiper.com/oauth-callback` as the callback URL. Please note that Twitch requires this URL to use the `HTTPS` schema.
+
+image::identity-providers/twitch-create-app.png[Create Twitch App,width=1200,role=shadowed]
+
+=== Choose a Category
+
+You can select the category of your application. Options include `Game Integration`, `Website Integration`, `Application Integration`, `Browser Extension`, `Broadcaster Suite`, `Chat Bot`, `Giveaway/Loyalty Tool`, `Analytics Tool`, `Other`.
+
+=== Keys and token
+
+Once you click create, you will be redirected back to the dashboard for applications. From there, you will have to click `Manage` to view the client id and the client secret. It is very likely you will need to create a secret by clicking on the "New Secret" button. These two values will be used to configure the FusionAuth Identity Provider.
+
+There is no further configuration to perform on this panel, you will only need to record or copy these values for the next step.
+
+Once you have completed this configuration you will be able to enable the Twitch login button for one or more FusionAuth Applications. Below is an example login page with the Twitch Identity Provider enabled.
+
+image::identity-providers/login-twitch.png[Twitch Login]
+
+== Create a Twitch Identity Provider
+
+:identity_provider: Twitch
+include::docs/v1/tech/identity-providers/_idp_application_configuration.adoc[]
+
+image::identity-providers/identity-provider-twitch-add.png[Twitch Home,width=1200]
+
+:get_started_link: link:https://dev.twitch.tv/[Twitch - Getting Started]
+:scope_addon: The values 'openid' 'user:read:email' are defaulted for this IdP.
+:identity_provider: Twitch
+:linking_strategy: Create a pending link
+
+include::docs/v1/tech/identity-providers/_idp-form-fields.adoc[]
+
+:linking_strategy!:
+:identity_provider!:
+:scope_addon!:
+:get_started_link!:
++
++
+
+== Try a Login
+
+Twitch may ask on the login if you intend to redirect to your FusionAuth application.
```

Comment:
```suggestion
When a user logs in, Twitch may warn about the redirection to your FusionAuth application.
```

Does this happen every time? Is it only because we're in a development mode?

Reply:
Not entirely sure.  It's a single-time event until the cookies are cleared (I think).  I would actually have to do some more testing.